Frontend/src/Components/Battery/Battery.js

```diff
@@ -0,0 +1,140 @@
+import {Flex, Center, Box, Spacer, Text, useColorMode} from "@chakra-ui/react";
+import RangeRow from "../Shared/RangeRow";
+import RangeBar from "../Shared/RangeBar";
+import CONSTANTS from "../../data-constants.json";
+import getColor from "../Shared/colors";
+
+export default function Battery(props) {
+    const { colorMode } = useColorMode();
+    const headerBg = getColor("header", colorMode);
+    const borderColor = getColor("border", colorMode);
+
+    const fanEnableText = (props.data?.fan_enable[0] ?? -1.0) ? 'ON' : 'OFF';
+    const fanEnableColor = (props.data?.fan_enable[0] ?? -1.0) ? '#89FC4D' : '#F5C2C1';
+
+    const chargeEnableText = (props.data?.charge_enable[0] ?? -1.0) ? 'Enable' : 'Disable';
+    const chargeEnableColor = (props.data?.charge_enable[0] ?? -1.0) ? '#89FC4D' : '#F5C2C1';
+
+    return (
+        <Flex flex="auto" direction="column" overflowY="scroll">
+            <RangeRow
+                dataTitle="Pack Temp"
+                dataValue={props.data?.pack_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_temp}
+            />
+            <RangeRow
+                dataTitle="Pack Internal Temp"
+                dataValue={props.data?.pack_internal_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_internal_temp}
+
+                bg={headerBg}
+            />
+            <Center
+                borderTopWidth={1}
+                borderBottomWidth={1}
+                borderRightWidth={1}
+                borderColor={borderColor}
+                lineHeight='1.2em'
+                height='100%'
+            >
+                <Flex w='95%'>
+                    <Flex direction='column'>
+                        <Spacer />
+                        <Box
+                            border={"1px"}
+                            borderRadius='0px'
+                            h='1em'
+                            background={fanEnableColor}
+                        >
+                            <Text fontSize='xs' margin='0.05px'>{fanEnableText}</Text>
+                        </Box>
+                        <Spacer />
+                    </Flex>
```

Comment:
Rather than using the `Flex`s and `Spacer`s, try using a percentage height and vertical margins for the Box. To center the text, let's use a Center rather than a Box. Also, can you please make the text bold and add some padding inside the Center (or increase the margin around the Text) so that the text doesn't touch the border? Additionally, can you please make the Center's border the same width as the RangeBars?

Reply:
Made these changes, but you'll have to let me know whether you think some of the numerical values I chose make sense on your screen.